### PR TITLE
Post 7.2.3.ga cp to release sync

### PR DESCRIPTION
**Pull request by iankko**
Sync the changes from `sso72-cp` branch to content of `sso72` branch post the RH-SSO 7.2.3.GA release. Out of all [these](https://github.com/jboss-container-images/redhat-sso-7-openshift-image/compare/sso72...sso72-cp) changes, we just need the following four:

1. [Import of ARQ tests](https://github.com/jboss-container-images/redhat-sso-7-openshift-image/commit/931c74494d2b99266d3117608dd8e3fa1f0aba01)
2. [Bump RH-SSO for OpenShift image version to "1.2"](https://github.com/jboss-container-images/redhat-sso-7-openshift-image/commit/cef7cb5fd009596881452382fe79dfd5e465e8ed)
3. [CLOUD-2413 move material out to modules ](https://github.com/jboss-container-images/redhat-sso-7-openshift-image/commit/be5eeef6a358b013f87ebd63c5dc1ca8b0c6c2b2)
4. [use sprint-19 changes ](https://github.com/jboss-container-images/redhat-sso-7-openshift-image/commit/7020a7c214e2b4fc04bf7a879e7d4eab31f818af) -- this one was updated to perform the change against the `image.yaml` file, instead of to `override-cp.yaml`, which is not present in `sso72` branch

The remaining ones aren't needed.

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by iankko** [comment]
@ASzc @rcernich @goldmann PTAL (once got a chance)

Thank you, Jan

**Review by ASzc — COMMENTED** [review]
So, the new commit for sprint 19 was made after the build. I guess it's ok if the resulting image.yaml is the same, but this seems like something the overrides files were supposed to avoid.

**Comment by iankko** [comment]
What / how do you suggest to deal with this? Should we introduce an override file for release branch too? (sounds artificial, since its a release branch from the beginning, right?)

The current approach (make a patch after actual release) isn't ideal, but IMHO hand-editing / diffutils editing of the original patch to work against ```image.yaml``` file, instead of ```override-cp.yaml``` file is even worse one (since I would edit change originally done by someone else against completely different file, just to get it working properly)

**Comment by goldmann** [comment]
TL;DR; It's fine to create a post-merge commit to update the ref.

General idea is that the image source is correct at any time. At release time code is synced between branches. This is all we know.

Now, there are things in the source code (especially the image descriptor) which contains parts that are always different between branches. The suggestion was always to use overrides files to define bits that are different between branches (`osbs` section, for example). (Ping me to get link to the source)

On the other hand I think I never mentioned module references and how these should be used. I don't think there is a solution in the current workflow that would end up in seamless merging between branches. Always you will need to double check if the merge was done correct and if the content is what you expect.

One way of making it simpler would be to use [git attributes to define merge strategies](https://git-scm.com/book/en/v2/Customizing-Git-Git-Attributes#_merge_strategies). To implement it you would need to create a `.gitattributes` file with content like this:

```properties
*override*.yaml merge=ours
```

And then:

```
$ git config --global merge.ours.driver true
```

This would instruct Git that you want to actually ignore changes made to override files. This would prevent any merge conflicts for these files. This **does not mean** that you would be not required to double check if the override file is in correct state. This implies that after you merge, you should check that the references are correct.

I'm not very happy about this and we need to find a better solution for it. Comments are welcome, please send proposals to mailing list.

**Comment by iankko** [comment]
@goldmann 

Thanks, Marek! Would it solve the issue, if instead of using ```master``` in image.yaml, and ```sprint-*``` in override-cp.yaml file in sso72-cp branch, we would do it vice versa way? IOW the image.yaml would have cct_module reference set to last released image version ```sprint-19``` in this case. The override.yaml would have ```master```.

This way, if the build was needed for CP update in the current Sprint, the engineer building the image would use ```override-cp.yaml``` file. In the moment image is approved for release and respective cct_module tag exists, a PR would be issued, to update cct_module reference in image.yaml to match the expected value.

After the release, such PR could be smoothly merged into release branch too?

WDYT?

**Comment by iankko** [comment]
Also, can I merge the current PR as is (my understanding its allowed to issue a commit even after a release, if it brings the release back to the stable state), or are there any other changes needed?

**Comment by iankko** [comment]
Merging to be able to build the image. The proper approach how to smooth down the ```sso72-cp``` to ```sso72``` branches sync to be discussed yet.

**Comment by rcernich** [comment]
I think the correct solution here is to use the release overrides when doing the sprint builds and update it to point to the correct sprint branch prior to doing the sprint builds.  This ensures everything stays in sync and our builds can be reproduced.  This may mean the release overrides points to one branch in -cp and cve branches and a newer branch in dev.  With the new --overrides-file option in cekit 2.0, it may be worth investigating the best way to organize the data, e.g. maybe one file to pin cct_modules, another to manage the differences between dev/cp/release base image, etc.
